mlflow/gateway/utils.py

```diff
@@ -80,3 +82,14 @@ def get_gateway_uri() -> str:
             f"`mlflow.set_gateway_uri()` or set the environment variable {MLFLOW_GATEWAY_URI.name} "
             "to the running Gateway API server's uri"
         )
+
+
+def assemble_uri_path(parts: List[str]) -> str:
+    """
+    Assemble a correct URI path from a list of path parts.
+
+    :param parts: A list of strings representing parts of a URI path.
+    :return: A string representing the complete assembled URI path.
+    """
+    path_parts = [part for path in parts for part in path.strip("/").split("/") if part]
+    return "/" + "/".join(path_parts)
```

Comment:
/ai any better implementation for this function?

Reply:
@harupy can you ask /ai if there are any risks for using `os.path` here instead of `posixpath` ;)

Reply:
/ai are there any risks for using `os.path` here instead of `posixpath` in this function?

Reply:
Hmmmm /ai.... :P

Reply:
Let me ask the same question to GPT-4.

Reply:
@dbczumar 

<img width="860" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/b117bc75-7185-41a9-a69e-2484adc0aabb">

Reply:
GPT-4 knows but GPT-3.5 doesn't

Reply:
Awesome, hope we can use GPT-4 when we open this up :P

Reply:
+1. I feel like an argument to specify the model seems useful.


```
/ai --model gpt-4 ...
```